chunk/aws_storage_client_test.go

```diff
@@ -101,11 +101,43 @@ func (m *mockDynamoDBClient) queryRequest(input *dynamodb.QueryInput) dynamoDBRe
 		Items: []map[string]*dynamodb.AttributeValue{},
 	}
 
+	// Required filters
 	hashValue := *input.KeyConditions[hashKey].AttributeValueList[0].S
-	items := m.tables[*input.TableName].items[hashValue]
 
-	// TODO we should also filter by range value
+	// Optional filters
+	var (
+		rangeValueFilter     []byte
+		rangeValueFilterType string
+		valueFilter          []byte
+		valueFilterType      *string
+	)
+	if input.KeyConditions[rangeKey] != nil {
```

Comment:
It might be more idiomatic to do:

```go
if c, ok := input.KeyConditions[rangeKey]; ok {
    rangeValueFilter = c.AttributeValueList[0].B
    rangeValueFilterType = *c.ComparisonOperator
}
```